### ISSUE TITLE
[JENKINS-54589] Builds are failing with timeout during checkout. Issue will be fixed only after the server restart where Jenkins is installed. Intermittent issue but happening frequently. Please assist on this.

### ISSUE THREAD

**Issue by jenkins-infra-bot** [jenkinsci/git-plugin#3360]
<p>Builds are failing with timeout during checkout. Issue will be fixed only after the server restart where Jenkins is installed. Intermittent issue but happening frequently. Please assist on this.</p>

```
 
> git.exe config core.sparsecheckout # timeout=10
*09:41:33*  
> git.exe checkout -f ***
*09:51:33* ERROR: Timeout after 10 minutes
*09:51:33* FATAL: Could not checkout ***
*09:51:33* hudson.plugins.git.GitException: Command "git.exe checkout -f ***" returned status code -1:
*09:51:33* stdout: 
*09:51:33* stderr: 
*09:51:33* 	at org.jenkinsci.plugins.gitclient.CliGitAPIImpl.launchCommandIn(CliGitAPIImpl.java:1990)
*09:51:33* 	at org.jenkinsci.plugins.gitclient.CliGitAPIImpl.access$800(CliGitAPIImpl.java:72)
*09:51:33* 	at org.jenkinsci.plugins.gitclient.CliGitAPIImpl$9.execute(CliGitAPIImpl.java:2289)

```

---
<details><summary><i>Originally reported by <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=tejasgr">tejasgr</a>, imported from: <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-54589" target="_blank">Builds are failing with timeout during checkout. Issue will be fixed only after the server restart where Jenkins is installed. Intermittent issue but happening frequently. Please assist on this.</a></i></summary>
<i><ul>
<li><b>status</b>: Closed
<li><b>priority</b>: Major
<li><b>component(s)</b>: git-plugin
<li><b>resolution</b>: Not A Defect
<li><b>resolved</b>: 2018-11-13T16:41:53+00:00
<li><b>votes</b>: 1
<li><b>watchers</b>: 1
<li><b>imported</b>: 2025-12-02
</ul></i>
<details><summary>Raw content of original issue</summary>

<pre>
<p>Builds are failing with timeout during checkout. Issue will be fixed only after the server restart where Jenkins is installed. Intermittent issue but happening frequently. Please assist on this.</p>

<div class="preformatted panel" style="border-width: 1px;"><div class="preformattedContent panelContent">
<pre> 
&gt; git.exe config core.sparsecheckout # timeout=10
*09:41:33*  
&gt; git.exe checkout -f ***
*09:51:33* ERROR: Timeout after 10 minutes
*09:51:33* FATAL: Could not checkout ***
*09:51:33* hudson.plugins.git.GitException: Command "git.exe checkout -f ***" returned status code -1:
*09:51:33* stdout: 
*09:51:33* stderr: 
*09:51:33* 	at org.jenkinsci.plugins.gitclient.CliGitAPIImpl.launchCommandIn(CliGitAPIImpl.java:1990)
*09:51:33* 	at org.jenkinsci.plugins.gitclient.CliGitAPIImpl.access$800(CliGitAPIImpl.java:72)
*09:51:33* 	at org.jenkinsci.plugins.gitclient.CliGitAPIImpl$9.execute(CliGitAPIImpl.java:2289)
</pre>
</div></div></pre>
</details>
</details>
<details><summary><i>1 attachment</i></summary>

- [image-2018-11-13-14-34-01-798.png](https://issues.jenkins.io/secure/attachment/45149/image-2018-11-13-14-34-01-798.png)
  > ![image-2018-11-13-14-34-01-798.png](https://issues.jenkins.io/secure/attachment/45149/image-2018-11-13-14-34-01-798.png)
</details>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/markewaite.png" title="markewaite's avatar" /> markewaite:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-54589?focusedId=353613&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-353613">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>This is not a bug.  Please don't use bug reports to request help with configuration issues.  Very few people read bug reports for specific plugins, and even fewer respond to requests for help that arrive as bug reports.</p>

<p>Please use the Jenkins chat channels (IRC or gitter) or the Jenkins users mailing list to request help.  There are many more users in those locations that can assist.</p>
</pre>
</details>

</ul>
</details>

<p>This is not a bug.  Please don't use bug reports to request help with configuration issues.  Very few people read bug reports for specific plugins, and even fewer respond to requests for help that arrive as bug reports.</p>

<p>Please use the Jenkins chat channels (IRC or gitter) or the Jenkins users mailing list to request help.  There are many more users in those locations that can assist.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/annetheagile.png" title="annetheagile's avatar" /> annetheagile:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-54589?focusedId=353694&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-353694">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>This is actually a bug. It needs to be clarified to show that. </p>

<p>I have jobs that are failing git steps within a minute. For example, today, (with variables for the git commit hash and repo):</p>
<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-java">05:39:46  &gt; git config core.sparsecheckout # timeout=10
05:39:46  &gt; git checkout -f &lt;GITSHA&gt; # timeout=4
05:39:48 Commit message: <span class="code-quote">"Merge pull request #71 from MYORG/MYTICKET-3"</span>
05:39:48  &gt; git rev-list --no-walk &lt;GITSHA&gt; # timeout=10
05:39:48 java.util.concurrent.TimeoutException: Timeout waiting <span class="code-keyword">for</span> task.
05:39:48 	at com.google.common.util.concurrent.AbstractFuture$Sync.get(AbstractFuture.java:259)
05:39:48 	at com.google.common.util.concurrent.AbstractFuture.get(AbstractFuture.java:91)
05:39:48 	at com.google.common.util.concurrent.ForwardingFuture.get(ForwardingFuture.java:69)
05:39:48 	at com.atlassian.jira.<span class="code-keyword">rest</span>.client.internal.async.DelegatingPromise.get(DelegatingPromise.java:113)
05:39:48 	at hudson.plugins.jira.JiraRestService.getIssuesFromJqlSearch(JiraRestService.java:194)
05:39:48 	at hudson.plugins.jira.JiraSession.getIssuesFromJqlSearch(JiraSession.java:135)
05:39:48 	at io.jenkins.blueocean.service.embedded.jira.JiraSCMListener.onChangeLogParsed(JiraSCMListener.java:43)
05:39:48 	at hudson.model.listeners.SCMListener.onChangeLogParsed(SCMListener.java:120)
05:39:48 	at hudson.model.AbstractBuild$AbstractBuildExecution.defaultCheckout(AbstractBuild.java:590)
05:39:48 Caused: java.io.IOException: Failed to parse changelog
05:39:48 	at hudson.model.AbstractBuild$AbstractBuildExecution.defaultCheckout(AbstractBuild.java:592)
05:39:48 	at jenkins.scm.SCMCheckoutStrategy.checkout(SCMCheckoutStrategy.java:86)
05:39:48 	at hudson.model.AbstractBuild$AbstractBuildExecution.run(AbstractBuild.java:499)
05:39:48 	at hudson.model.Run.execute(Run.java:1810)
05:39:48 	at hudson.model.FreeStyleBuild.run(FreeStyleBuild.java:43)
05:39:48 	at hudson.model.ResourceController.execute(ResourceController.java:97)
05:39:48 	at hudson.model.Executor.run(Executor.java:429)
05:39:48 [PostBuildScript] - Executing post build scripts.
05:39:48 [ios-DAH] $ /bin/bash -l /<span class="code-keyword">var</span>/folders/0r/wr2c4hs13r58l9k7pjdxpnl80000gq/T/jenkins8422288554445462114.sh
 </pre>
</div></div>
<p>^^ This only just started after I upgraded jenkins/plugins last night.</p>

<p>Another one today,  also taking 1 ms to timeout;</p>
<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-java">10:03:23  &gt; git rev-list --no-walk &lt;GITSHA&gt; # timeout=10
10:03:24 java.util.concurrent.TimeoutException: Timeout waiting <span class="code-keyword">for</span> task.
 </pre>
</div></div>
<p> </p>

<p> </p>

<p>At first I thought the issue was lack of specifically set timeouts so I added git advanced settings for advanced/checkout-options and advanced/clone-options to make timeout 4minutes. That seemed to work, but it did not and is intermittent , as mentioned by OP. </p>

<p> </p>

<p>I am on; </p>

<p>Enterprise Github 2.12.2 (per tooltip on footer/octocat)</p>

<p>Jenkins ver. 2.151 (after previously was 2.144 and was ok)</p>
<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-java"> Git plugin  3.9.1 
and others such as  Git Changelog  2.11   Git client plugin 2.7.3     Git Pipeline <span class="code-keyword">for</span> Blue Ocean  1.8.4    GIT server Plugin  1.7 GitHub API Plugin  1.92   GitHub Branch Source Plugin  2.4.1 </pre>
</div></div>
<p>^^ not sure which are the most important , so tried to paste a subset of plugins.</p>
</pre>
</details>

</ul>
</details>

<p>This is actually a bug. It needs to be clarified to show that. </p>

<p>I have jobs that are failing git steps within a minute. For example, today, (with variables for the git commit hash and repo):</p>

<pre>
05:39:46  > git config core.sparsecheckout # timeout=10
05:39:46  > git checkout -f <GITSHA> # timeout=4
05:39:48 Commit message: <span class="code-quote">"Merge pull request #71 from MYORG/MYTICKET-3"</span>
05:39:48  > git rev-list --no-walk <GITSHA> # timeout=10
05:39:48 java.util.concurrent.TimeoutException: Timeout waiting <span class="code-keyword">for</span> task.
05:39:48 	at com.google.common.util.concurrent.AbstractFuture$Sync.get(AbstractFuture.java:259)
05:39:48 	at com.google.common.util.concurrent.AbstractFuture.get(AbstractFuture.java:91)
05:39:48 	at com.google.common.util.concurrent.ForwardingFuture.get(ForwardingFuture.java:69)
05:39:48 	at com.atlassian.jira.<span class="code-keyword">rest</span>.client.internal.async.DelegatingPromise.get(DelegatingPromise.java:113)
05:39:48 	at hudson.plugins.jira.JiraRestService.getIssuesFromJqlSearch(JiraRestService.java:194)
05:39:48 	at hudson.plugins.jira.JiraSession.getIssuesFromJqlSearch(JiraSession.java:135)
05:39:48 	at io.jenkins.blueocean.service.embedded.jira.JiraSCMListener.onChangeLogParsed(JiraSCMListener.java:43)
05:39:48 	at hudson.model.listeners.SCMListener.onChangeLogParsed(SCMListener.java:120)
05:39:48 	at hudson.model.AbstractBuild$AbstractBuildExecution.defaultCheckout(AbstractBuild.java:590)
05:39:48 Caused: java.io.IOException: Failed to parse changelog
05:39:48 	at hudson.model.AbstractBuild$AbstractBuildExecution.defaultCheckout(AbstractBuild.java:592)
05:39:48 	at jenkins.scm.SCMCheckoutStrategy.checkout(SCMCheckoutStrategy.java:86)
05:39:48 	at hudson.model.AbstractBuild$AbstractBuildExecution.run(AbstractBuild.java:499)
05:39:48 	at hudson.model.Run.execute(Run.java:1810)
05:39:48 	at hudson.model.FreeStyleBuild.run(FreeStyleBuild.java:43)
05:39:48 	at hudson.model.ResourceController.execute(ResourceController.java:97)
05:39:48 	at hudson.model.Executor.run(Executor.java:429)
05:39:48 [PostBuildScript] - Executing post build scripts.
05:39:48 [ios-DAH] $ /bin/bash -l /<span class="code-keyword">var</span>/folders/0r/wr2c4hs13r58l9k7pjdxpnl80000gq/T/jenkins8422288554445462114.sh
 </pre>
<p>^^ This only just started after I upgraded jenkins/plugins last night.</p>

<p>Another one today,  also taking 1 ms to timeout;</p>

<pre>
10:03:23  > git rev-list --no-walk <GITSHA> # timeout=10
10:03:24 java.util.concurrent.TimeoutException: Timeout waiting <span class="code-keyword">for</span> task.
 </pre>
<p> </p>

<p> </p>

<p>At first I thought the issue was lack of specifically set timeouts so I added git advanced settings for advanced/checkout-options and advanced/clone-options to make timeout 4minutes. That seemed to work, but it did not and is intermittent , as mentioned by OP. </p>

<p> </p>

<p>I am on; </p>

<p>Enterprise Github 2.12.2 (per tooltip on footer/octocat)</p>

<p>Jenkins ver. 2.151 (after previously was 2.144 and was ok)</p>

<pre>
 Git plugin  3.9.1 
and others such as  Git Changelog  2.11   Git client plugin 2.7.3     Git Pipeline <span class="code-keyword">for</span> Blue Ocean  1.8.4    GIT server Plugin  1.7 GitHub API Plugin  1.92   GitHub Branch Source Plugin  2.4.1 </pre>
<p>^^ not sure which are the most important , so tried to paste a subset of plugins.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/annetheagile.png" title="annetheagile's avatar" /> annetheagile:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-54589?focusedId=353696&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-353696">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>1. </p>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=tejasgr" class="user-hover" rel="tejasgr">tejasgr</a> please let us know what versions you are on, and if you have enterprise github too. </p>

<p>Also you said restart would fix the issue , do you mean it would be completely failing until you restart? I do not have that, I have intermittent and I have not rebooted to see if that changed it.</p>

<p>2.</p>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=markewaite" class="user-hover" rel="markewaite">markewaite</a> I am confused deciding which plugin is the cause of things like this. From this line I decided that regular git might be the cause; </p>
<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-java">05:39:48 	at jenkins.scm.SCMCheckoutStrategy.checkout(SCMCheckoutStrategy.java:86) </pre>
</div></div>
<p>Am I right?</p>
</pre>
</details>

</ul>
</details>

<p>1. </p>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=tejasgr" class="user-hover" rel="tejasgr">tejasgr</a> please let us know what versions you are on, and if you have enterprise github too. </p>

<p>Also you said restart would fix the issue , do you mean it would be completely failing until you restart? I do not have that, I have intermittent and I have not rebooted to see if that changed it.</p>

<p>2.</p>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=markewaite" class="user-hover" rel="markewaite">markewaite</a> I am confused deciding which plugin is the cause of things like this. From this line I decided that regular git might be the cause; </p>

<pre>
05:39:48 	at jenkins.scm.SCMCheckoutStrategy.checkout(SCMCheckoutStrategy.java:86) </pre>
<p>Am I right?</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/annetheagile.png" title="annetheagile's avatar" /> annetheagile:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-54589?focusedId=353698&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-353698">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Hmm, since my timeouts are within 1ms maybe I should have a separate ticket and move my thoughts there?</p>

<p>The OP timeout was long.</p>
</pre>
</details>

</ul>
</details>

<p>Hmm, since my timeouts are within 1ms maybe I should have a separate ticket and move my thoughts there?</p>

<p>The OP timeout was long.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>tejasgr:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-54589?focusedId=353701&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-353701">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Yes. I am also on Enterprise Git 2.15.1. Untill remote server restart where Jenkins installed, the issue will be there. Only with remote server restart where Jenkins installed, the issue will be cleared and again facing it sometime like the next day or back sometime.</p>

<p> </p>
</pre>
</details>

</ul>
</details>

<p>Yes. I am also on Enterprise Git 2.15.1. Untill remote server restart where Jenkins installed, the issue will be there. Only with remote server restart where Jenkins installed, the issue will be cleared and again facing it sometime like the next day or back sometime.</p>

<p> </p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/markewaite.png" title="markewaite's avatar" /> markewaite:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-54589?focusedId=353702&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-353702">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=annetheagile" class="user-hover" rel="annetheagile">annetheagile</a> the stack trace that you are reporting is dramatically different from the stack trace reported in this bug.</p>

<p>Your stack trace (with my subsetting):</p>

<div class="preformatted panel" style="border-width: 1px;"><div class="preformattedContent panelContent">
<pre>05:39:48 java.util.concurrent.TimeoutException: Timeout waiting for task.
05:39:48 	at com.google.common.util.concurrent.AbstractFuture$Sync.get(AbstractFuture.java:259)
05:39:48 	at com.google.common.util.concurrent.AbstractFuture.get(AbstractFuture.java:91)
05:39:48 	at com.google.common.util.concurrent.ForwardingFuture.get(ForwardingFuture.java:69)
05:39:48 	at com.atlassian.jira.rest.client.internal.async.DelegatingPromise.get(DelegatingPromise.java:113)
05:39:48 	at hudson.plugins.jira.JiraRestService.getIssuesFromJqlSearch(JiraRestService.java:194)
05:39:48 	at hudson.plugins.jira.JiraSession.getIssuesFromJqlSearch(JiraSession.java:135)
05:39:48 	at io.jenkins.blueocean.service.embedded.jira.JiraSCMListener.onChangeLogParsed(JiraSCMListener.java:43)
05:39:48 	at hudson.model.listeners.SCMListener.onChangeLogParsed(SCMListener.java:120)
05:39:48 	at hudson.model.AbstractBuild$AbstractBuildExecution.defaultCheckout(AbstractBuild.java:590)
</pre>
</div></div>

<p>seems to indicate that Blue Ocean is performing a JQL query to the Jira service to request the list of issues from the Jira service.  That query seems to be timing out.</p>

<p>I recommend you either begin a discussion on chat or the mailing list to help identify why your Jira server times out on that request, or open a ticket where those discussions can occur.  If you choose to use a ticket, you accept that fewer people will see your request and thus fewer people are likely able to help you.</p>
</pre>
</details>

</ul>
</details>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=annetheagile" class="user-hover" rel="annetheagile">annetheagile</a> the stack trace that you are reporting is dramatically different from the stack trace reported in this bug.</p>

<p>Your stack trace (with my subsetting):</p>



```
05:39:48 java.util.concurrent.TimeoutException: Timeout waiting for task.
05:39:48 	at com.google.common.util.concurrent.AbstractFuture$Sync.get(AbstractFuture.java:259)
05:39:48 	at com.google.common.util.concurrent.AbstractFuture.get(AbstractFuture.java:91)
05:39:48 	at com.google.common.util.concurrent.ForwardingFuture.get(ForwardingFuture.java:69)
05:39:48 	at com.atlassian.jira.rest.client.internal.async.DelegatingPromise.get(DelegatingPromise.java:113)
05:39:48 	at hudson.plugins.jira.JiraRestService.getIssuesFromJqlSearch(JiraRestService.java:194)
05:39:48 	at hudson.plugins.jira.JiraSession.getIssuesFromJqlSearch(JiraSession.java:135)
05:39:48 	at io.jenkins.blueocean.service.embedded.jira.JiraSCMListener.onChangeLogParsed(JiraSCMListener.java:43)
05:39:48 	at hudson.model.listeners.SCMListener.onChangeLogParsed(SCMListener.java:120)
05:39:48 	at hudson.model.AbstractBuild$AbstractBuildExecution.defaultCheckout(AbstractBuild.java:590)

```

<p>seems to indicate that Blue Ocean is performing a JQL query to the Jira service to request the list of issues from the Jira service.  That query seems to be timing out.</p>

<p>I recommend you either begin a discussion on chat or the mailing list to help identify why your Jira server times out on that request, or open a ticket where those discussions can occur.  If you choose to use a ticket, you accept that fewer people will see your request and thus fewer people are likely able to help you.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/markewaite.png" title="markewaite's avatar" /> markewaite:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-54589?focusedId=353703&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-353703">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Reaffirming this is not a bug.  The `git checkout` command can have its timeout value adjusted if needed for long checkout (due to a slow file system or a network file system or ...).  That's a configuration change, not a bug.</p>

<p>The `git checkout` command is entirely local.  It copies content from the local .git repository into the local working directory.  It is not performing a network fetch or other network operation of any sort.  It is not performing any operations which require credentials (unless you're using git large file support).</p>
</pre>
</details>

</ul>
</details>

<p>Reaffirming this is not a bug.  The `git checkout` command can have its timeout value adjusted if needed for long checkout (due to a slow file system or a network file system or ...).  That's a configuration change, not a bug.</p>

<p>The `git checkout` command is entirely local.  It copies content from the local .git repository into the local working directory.  It is not performing a network fetch or other network operation of any sort.  It is not performing any operations which require credentials (unless you're using git large file support).</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>tejasgr:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-54589?focusedId=353704&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-353704">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Hi Mark,</p>

<p>As said its not actually timeout issue as per it says. After the restart, it even checksout within a minute. It always checksout within a minute. When this issue occurs, I did increased the timeout, changed braches and everything and it fails with same error. But once the server restart done, then everything will be fine and any branch checkout will be carried within a minute.</p>
</pre>
</details>

</ul>
</details>

<p>Hi Mark,</p>

<p>As said its not actually timeout issue as per it says. After the restart, it even checksout within a minute. It always checksout within a minute. When this issue occurs, I did increased the timeout, changed braches and everything and it fails with same error. But once the server restart done, then everything will be fine and any branch checkout will be carried within a minute.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/markewaite.png" title="markewaite's avatar" /> markewaite:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-54589?focusedId=353705&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-353705">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=tejasgr" class="user-hover" rel="tejasgr">tejasgr</a> when you say "remote server restart", do you mean that the GitHub Enterprise server restart seems to be required in order for the `git checkout` command to complete?  If not, can you explain further what server is restarted that causes things to be better behaved?</p>

<p>Are you using git large file support in the repository where you're performing that checkout?</p>

<p>Is the Jenkins workspace where the checkout is being performed on some form of network attached storage?  If so, is that storage performing reliably?</p>

<p>If on network storage, does the same problem occur if you use local storage instead of network attached storage?</p>
</pre>
</details>

</ul>
</details>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=tejasgr" class="user-hover" rel="tejasgr">tejasgr</a> when you say "remote server restart", do you mean that the GitHub Enterprise server restart seems to be required in order for the `git checkout` command to complete?  If not, can you explain further what server is restarted that causes things to be better behaved?</p>

<p>Are you using git large file support in the repository where you're performing that checkout?</p>

<p>Is the Jenkins workspace where the checkout is being performed on some form of network attached storage?  If so, is that storage performing reliably?</p>

<p>If on network storage, does the same problem occur if you use local storage instead of network attached storage?</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>tejasgr:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-54589?focusedId=353706&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-353706">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Hi Mark,</p>

<p>Its not github restart. We have jenkins service installed on a windows server. So when this issue occured, we had to restart this windows server where Jenkins is installed to get it working back again. Yes It is Jenkins workplace to where checkout needs to be performed from github.</p>
</pre>
</details>

</ul>
</details>

<p>Hi Mark,</p>

<p>Its not github restart. We have jenkins service installed on a windows server. So when this issue occured, we had to restart this windows server where Jenkins is installed to get it working back again. Yes It is Jenkins workplace to where checkout needs to be performed from github.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/markewaite.png" title="markewaite's avatar" /> markewaite:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-54589?focusedId=353708&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-353708">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Is the stack trace in the bug description accurately describing the stack trace that you are seeing when checkout fails to complete on your Windows server?  I assume it is.  If it is not, then please place an accurate stack trace in the bug report for better diagnosis.</p>

<p>If that stack trace is accurate, then there is a chance that a git process (either from Jenkins or elsewhere) has acquired a lock on the <tt>.git</tt> directory inside the Jenkins workspace.  That is not something I've ever seen before, but it is conceptually possible that another process might have a lock which causes the local git checkout to block while waiting for the lock to be released.</p>

<p>Next time the problem occurs, use a command prompt window or a git bash Window on the agent where the workspace resides.  Attempt to perform git operations (like `git ls-remote origin` and `git status` and `git log`) in that workspace while the Jenkins checkout is blocked.</p>
</pre>
</details>

</ul>
</details>

<p>Is the stack trace in the bug description accurately describing the stack trace that you are seeing when checkout fails to complete on your Windows server?  I assume it is.  If it is not, then please place an accurate stack trace in the bug report for better diagnosis.</p>

<p>If that stack trace is accurate, then there is a chance that a git process (either from Jenkins or elsewhere) has acquired a lock on the <tt>.git</tt> directory inside the Jenkins workspace.  That is not something I've ever seen before, but it is conceptually possible that another process might have a lock which causes the local git checkout to block while waiting for the lock to be released.</p>

<p>Next time the problem occurs, use a command prompt window or a git bash Window on the agent where the workspace resides.  Attempt to perform git operations (like `git ls-remote origin` and `git status` and `git log`) in that workspace while the Jenkins checkout is blocked.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/annetheagile.png" title="annetheagile's avatar" /> annetheagile:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-54589?focusedId=353714&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-353714">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>ty so much  <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=markewaite" class="user-hover" rel="markewaite">markewaite</a> ! I will investigate as you recommend.</p>

<p>FYI I'm using a Mac host for Jenkins server, so I guess many parts of my setup are indeed different from this ticket.</p>
</pre>
</details>

</ul>
</details>

<p>ty so much  <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=markewaite" class="user-hover" rel="markewaite">markewaite</a> ! I will investigate as you recommend.</p>

<p>FYI I'm using a Mac host for Jenkins server, so I guess many parts of my setup are indeed different from this ticket.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>tejasgr:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-54589?focusedId=353722&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-353722">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>It did started the block again. Failing while checkout. I have checked the directory and tried to checkout from git manually from the directory. But got this error. As you said. Yes. It is blocked. But why ut happens? And what would be the resolution for this? Its happening again and again</p>

<p><span class="image-wrap" style=""><img src="https://issues.jenkins.io/secure/attachment/45149/45149_image-2018-11-13-14-34-01-798.png" style="border: 0px solid black" /></span></p>
</pre>
</details>

</ul>
</details>

<p>It did started the block again. Failing while checkout. I have checked the directory and tried to checkout from git manually from the directory. But got this error. As you said. Yes. It is blocked. But why ut happens? And what would be the resolution for this? Its happening again and again</p>

<p><span class="image-wrap" style=""><img src="https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-last/refs/heads/main/attachments/45149/image-2018-11-13-14-34-01-798.png" style="border: 0px solid black" /></span></p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>tejasgr:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-54589?focusedId=353723&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-353723">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>But even if we remove lock file and start the build same issue at checkout. timeout and fails</p>
</pre>
</details>

</ul>
</details>

<p>But even if we remove lock file and start the build same issue at checkout. timeout and fails</p>
